### DISCUSSION TITLE
What is the logic behind -proxy [file] flag ,

### DISCUSSION THREAD

**Discussion by osamahamad** [projectdiscovery/nuclei#5320] [Q&A]
Hi, 

I have a question, I would like to know the logic behind this flag in this manner. 
What is the logic behind -proxy [file] flag, how it is gonna behave when running on a large hosts inputs list with a 1k socks5 proxy file. 
I have a list of 1k socks5 proxies, all verified and working, with no bandwidth limit and unlimited concurrency. 

When I provide -proxy [file] flag and the [file] has all 1k socks5 proxies mentioned above one per line. 

Is nuclei gonna use the first one for all the requests sent to the targets list and use second one only if the first line proxy is not working ? or 
is it gonna use each proxy of the list for each request ?
is it gonna use all of them randomly ? 
is it gonna use the proxies based on the amount and the targets list amount ? 

I want to understand that , my main objective is trying to rotate those proxies on each request made via nuclei on the targets list but it seems does not behave like that, it looks like it only use one proxy out of the 1k provided on the targets even after running it couble of times on the list ( checked by a basic way - adding a host i control with a listening port so i can capture the request sent via nuclei )

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
